src/AutoComplete/AutoComplete.js

```diff
@@ -265,21 +264,29 @@ class AutoComplete extends Component {
 
     const index = parseInt(child.key, 10);
     const chosenRequest = dataSource[index];
-    const searchText = this.chosenRequestText(chosenRequest);
-
-    this.setState({
-      searchText: searchText,
-    }, () => {
-      this.props.onUpdateInput(searchText, this.props.dataSource, {
-        source: 'touchTap',
-      });
-
+    if (typeof this.props.searchText !== 'undefined') {
       this.timerTouchTapCloseId = setTimeout(() => {
         this.timerTouchTapCloseId = null;
         this.close();
         this.props.onNewRequest(chosenRequest, index);
       }, this.props.menuCloseDelay);
-    });
+    } else {
+      const searchText = this.chosenRequestText(chosenRequest);
+
+      this.setState({
+        searchText: searchText,
+      }, () => {
+        this.props.onUpdateInput(searchText, this.props.dataSource, {
```

Comment:
You mean why we don't call `onUpdateInput` when `searchText` is controlled? If so, it's seems logical for me to not call it in this case (because we don't change input actually). If user wants to change `searchText`, he will do it in `onNewRequest` handler.